src/pages/settings/Profile/DisplayNamePage.js

```diff
@@ -57,60 +55,25 @@ class DisplayNamePage extends Component {
      */
     validate(values) {
         const errors = {};
-
-        // Check for invalid characters in first and last name
-        const [firstNameInvalidCharacter, lastNameInvalidCharacter] = ValidationUtils.findInvalidSymbols(
+        const [doesFirstNameHaveInvalidCharacters, doesLastNameHaveInvalidCharacters] = ValidationUtils.doesContainCommaOrSemicolon(
             [values.firstName, values.lastName],
         );
-        this.assignError(
-            errors,
-            'firstName',
-            !_.isEmpty(firstNameInvalidCharacter),
-            Localize.translateLocal(
-                'personalDetails.error.hasInvalidCharacter',
-                {invalidCharacter: firstNameInvalidCharacter},
-            ),
-        );
-        this.assignError(
-            errors,
-            'lastName',
-            !_.isEmpty(lastNameInvalidCharacter),
-            Localize.translateLocal(
-                'personalDetails.error.hasInvalidCharacter',
-                {invalidCharacter: lastNameInvalidCharacter},
-            ),
-        );
-        if (!_.isEmpty(errors)) {
-            return errors;
+
+        // First we validate the first name field
+        if (doesFirstNameHaveInvalidCharacters) {
+            errors.firstName = this.props.translate('personalDetails.error.hasInvalidCharacter');
+        } else if (ValidationUtils.doesContainReservedWord(values.firstName, CONST.DISPLAY_NAME.RESERVED_FIRST_NAMES)) {
+            errors.firstName = this.props.translate('personalDetails.error.containsReservedWord');
```

Comment:
What if the user did not input anything in the first name but added "Expensify" or "Concierge" on the last name? This make the user still have the name of reserved names.

<img width="379" alt="Screen Shot 2023-02-07 at 13 35 39" src="https://user-images.githubusercontent.com/25520267/217166799-fe8fff81-f29a-49ab-a3a9-c5951ae9875d.png">

Reply:
The back-end allows it, so I think it's fine.